src/main/java/com/aquarium/aquarium_backend/Controllers/AquariumController.java

```diff
@@ -2,39 +2,47 @@
 
 import com.aquarium.aquarium_backend.Services.AquariumService;
 import com.aquarium.aquarium_backend.databaseTables.Aquarium;
+import java.util.List;
+import java.util.Map;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
-
-import java.util.List;
-
+import org.springframework.web.bind.annotation.*;
 
 @RestController
 @RequestMapping(path = "api/v1/aquarium")
 public class AquariumController {
-    private final AquariumService aquariumService;
+  private final AquariumService aquariumService;
 
-    public AquariumController(AquariumService aquariumService) {
-        this.aquariumService = aquariumService;
-    }
+  public AquariumController(AquariumService aquariumService) {
+    this.aquariumService = aquariumService;
+  }
 
-    @GetMapping("/users/{userId}")
-    public ResponseEntity<?> getAquariumByUserId(@PathVariable Long userId) {
-        try {
-            List<Aquarium> aquariumList = aquariumService.getAllUsersAquariums(userId);
-            return new ResponseEntity<>(aquariumList, HttpStatus.OK);
-        }
-        catch (Exception e) {
-            return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
-        }
+  @GetMapping("/users/{userId}")
+  public ResponseEntity<?> getAquariumByUserId(@PathVariable Long userId) {
+    try {
+      List<Aquarium> aquariumList = aquariumService.getAllUsersAquariums(userId);
+      return new ResponseEntity<>(aquariumList, HttpStatus.OK);
+    } catch (Exception e) {
+      return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
     }
+  }
+
+  @GetMapping
+  public ResponseEntity<List<Aquarium>> getAllAquariums() {
+    List<Aquarium> aquariumList = aquariumService.getAllAquariums();
+    return new ResponseEntity<>(aquariumList, HttpStatus.OK);
+  }
 
-    @GetMapping
-    public ResponseEntity<List<Aquarium>> getAllAquariums() {
-        List<Aquarium> aquariumList = aquariumService.getAllAquariums();
-        return new ResponseEntity<>(aquariumList, HttpStatus.OK);
+  @PostMapping("add")
+  public ResponseEntity<?> addAquarium(@RequestBody Map<String, Object> payload) {
```

Comment:
You can pass Aquarium object, wiyhout ID insted of this ugly Map

Reply:
Aquarium has an object User inside which is unnecessary. In order to be consistent and not send user data over internet I would stick with current approach.